### PR TITLE
Customize the OpenTelemetry demo for Splunk

### PR DESCRIPTION
**Pull request by dmitchsplunk**
# Changes

I've made a number of customizations to the upstream OpenTelemetry demo for use with Splunk Cloud/Enterprise and Splunk Observability Cloud.  These include: 

- Replacing the upstream collector from the docker.yml file with the Splunk distro of the collector. 
- Removing the upstream collector from the kubernetes/opentelemetry-demo.yaml file (since it will be installed via helm instead), and modified the other services in this file to use the Splunk distro of the collector running on the K8s node. 
- Updated the README instructions to point Lantern blog articles for deploying the demo. 

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by jvoravong** [comment]
Can we develop a script ([example](https://github.com/signalfx/splunk-otel-collector-chart/blob/fb1a4af0ced365d2fff45484ed1f2164c2ae918e/examples/enable-operator-and-auto-instrumentation/update-demos.sh#L51)) or GitHub workflow to automate the synchronization of this demo with the upstream source? It's crucial to either automate the update process or provide detailed documentation for the process. This will facilitate incorporating changes from the upstream project over time.

**Comment by dmitchsplunk** [comment]
Thanks for the example script @jvoravong , I'll see what I can do.

**Comment by dmitchsplunk** [comment]
@jvoravong FYI, I've made good progress on automating the process of applying my changes to the upstream version, but still have work to do before the PR is ready to be reviewed.

**Comment by dmitchsplunk** [comment]
Hi @jvoravong - the requested changes are ready for your review.  

To make future merges with upstream straightforward, the only files changed are in a new folder named "splunk".  Everything else is aligned with upstream exactly. 

A new script was added named "update-demos.sh" that can be run after merging with upstream to regenerate the splunk-specific config.

File: splunk/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!-- markdownlint-disable-next-line -->
+
+## Welcome to the OpenTelemetry Astronomy Shop Demo
+
+This repository contains a fork of the OpenTelemetry Astronomy Shop, a microservice-based
+distributed system intended to illustrate the implementation of OpenTelemetry in
+a near real-world environment. It includes customizations for use with Splunk Observability Cloud.
+
+
```
Comment (jvoravong) [inline splunk/README.md]:
```suggestion
```

File: splunk/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Update Docker and Kubernetes Scripts
+
+After synchronizing changes with the upstream repository, the following command can be used to update the Splunk versions of the docker-compose.yml and kubernetes/opentelemetry-demo.yaml files, which are optimized for use with Splunk Observability Cloud: 
+
+````
```
Comment (jvoravong) [inline splunk/README.md]: IDEs can run these commands with useful tools if we modify sections like this slightly.
```suggestion
```bash
```

File: splunk/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+./update-demos.sh
+````
```
Comment (jvoravong) [inline splunk/README.md]:
```suggestion
```
```

File: splunk/update-demos.sh [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    # - name: OTEL_RESOURCE_ATTRIBUTES
+    #   value: service.name=$(OTEL_SERVICE_NAME),service.namespace=opentelemetry-demo,deployment.environment=development
+    yq eval -i ' (.. | select(tag == "!!str")) |= sub("service.namespace=opentelemetry-demo", "service.namespace=opentelemetry-demo,deployment.environment=development")'  "$SPLUNK_K8S_PATH"
+
+    echo "Completed updating the kubernetes/opentelemetry-demo.yaml for the OpenTelemetry demo app!"
+}
+
+# ---- OpenTelemetry Demo Update ----
+update_otel_demo_docker
+update_otel_demo_k8s
```
Comment (jvoravong) [inline splunk/update-demos.sh:129]: We should add one more function in this file. We should add a update_root_readme function that appends to or updates the root README.md.
- The root readme should have an obvious reference to the splunk/README.md
- We want users to know our Splunk resources exist in a certain location.

I can help out with this in a follow up PR if you don't get to it. This PR is already pretty big.

**Review by jvoravong — APPROVED** [review]
Some small nits. LGTM overall.

File: splunk/update-demos.sh (hunk 4 above)
Comment (dmitchsplunk) [reply splunk/update-demos.sh:129]: Great idea, thanks.  I've applied this change.

**Comment by dmitchsplunk** [comment]
Thanks @jvoravong , I've made the suggested changes.  I believe @atoulme is the owner of this repo and will need to approve the merge.

**Comment by atoulme** [comment]
@jvoravong you should have powers to merge now.

**Pull request merged**